elements/rh-dialog/rh-dialog.css

```diff
@@ -1,194 +1,227 @@
+/* NOTE: @deprecated, use `::backdrop` instead. Remove for RHDS v3. */
+#overlay {
+  background-color:
+    var(--rh-dialog-overlay-background-color,
+      var(--rh-dialog-backdrop-background-color));
+  block-size: 100%;
+  inset-block-start: 0;
+  inset-inline-start: 0;
   position: fixed;
-  height: 100%;
-  width: 100%;
-  top: 0;
-  left: 0;
-  background-color: rgba(3, 3, 3, 0.62);
+  inline-size: 100%;
 }
 
-[part='dialog'] {
-  position: relative;
-  margin: 0 auto;
-  width: var(--_box-width, calc(100% - var(--rh-space-2xl, 32px)));
-  max-height: var(--_box-max-height, calc(100% - var(--rh-space-3xl, 48px)));
-  box-shadow:
-    0 1rem 2rem 0 rgba(3, 3, 3, 0.16),
-    0 0 0.5rem 0 rgba(3, 3, 3, 0.1);
-  padding: var(--rh-space-xl, 24px);
-  margin-inline: var(--rh-space-lg, 16px);
-  background-color: var(--rh-color-surface-lightest, #ffffff);
-  max-width: min(90%, 1140px);
-  border-radius: var(--rh-border-radius-default, 3px);
-  color: var(--rh-color-text-primary-on-light, #151515);
-  font-family: inherit;
+::backdrop {
+  background-color: var(--rh-dialog-backdrop-background-color);
 }
 
-:host([width]) [part='dialog'],
-:host([variant]) [part='dialog'] {
-  margin-inline: 0;
+#dialog {
+  background-color: var(--rh-color-surface);
+  border-radius: var(--rh-border-radius-default, 3px);
+  border: 0;
+  box-shadow: var(--rh-box-shadow-xl, 0 8px 24px 3px rgba(21, 21, 21, 0.35));
+  box-sizing: border-box;
+  color: var(--rh-color-text-primary);
+  font-family: inherit;
+  inline-size: 100%;
+  margin-inline: auto;
+  max-block-size: var(--_box-max-block-size, calc(100vh - var(--rh-space-3xl, 48px)));
+  max-inline-size: var(--_box-width, min(90%, 1140px));
+  overflow-y: auto;
+  overscroll-behavior: contain;
+  padding: 0;
+  position: relative;
 }
 
-:host([width='small']) [part='dialog'],
-:host([variant='small']) [part='dialog'] {
+:host([width='small']) #dialog,
+:host([variant='small']) #dialog {
   --_box-width: 35rem;
 }
 
-:host([width='medium']) [part='dialog'],
-:host([variant='medium']) [part='dialog'] {
+:host([width='medium']) #dialog,
+:host([variant='medium']) #dialog {
   --_box-width: 52.5rem;
 }
 
-:host([width='large']) [part='dialog'],
-:host([variant='large']) [part='dialog'] {
+:host([width='large']) #dialog,
+:host([variant='large']) #dialog {
   --_box-width: 70rem;
 }
 
-[part='content'] {
-  overflow-y: auto;
-  overscroll-behavior: contain;
-  max-height: var(--_box-max-height, calc(100vh - var(--rh-space-3xl, 48px)));
-  box-sizing: border-box;
-  border-radius: var(--rh-border-radius-default, 3px);
-}
-
-[part='content'] ::slotted([slot='header']) {
-  margin-top: 0 !important;
+#content {
+  padding: var(--rh-space-xl, 24px);
+  margin-block-start: calc(-1 * var(--offset-top));
 }
 
-header {
+#header {
+  background-color: var(--rh-color-surface);
   position: sticky;
-  top: 0;
-  background-color: var(--rh-color-surface-lightest, #ffffff);
+  inset-block-start: calc(-1 * var(--offset-top));
```

Comment:
When viewport height is small the content in the dialog modal is hard to read, maybe undo the `position: sticky` for vertically challenged viewports.

```Suggestion
  inset-block-start: calc(-1 * var(--offset-top));
  @media (max-height: 320px) {
    position: unset;
  }
```

Reply:
You can also invert that rule starting with the position: unset, and applying sticky on`min-width: 320px`.  mobile first :)

Reply:
Good suggestion. I was not thinking about that.

Implemented this in [3b9e903](https://github.com/RedHat-UX/red-hat-design-system/pull/2078/commits/3b9e9033d8c7a04bc2f795cb308a058fa564f643). I didn't think `320px` was adequate, so I stepped up the media query a fair amount. So it should only happen if there's (hopefully) enough space.

If, at this point, it doesn't work for someone, they can override using shadow parts and go up or down in media query pixel width/height as needed.